### PR TITLE
feat(html): Add SimpleTable.ts

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
I'm using it in the DataPatterns output, but evidently something semi-recently changed RE package source loading in the gallery?

### CODE REVIEW

File: packages/html/src/SimpleTable.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    enter(domNode, element) {
+        super.enter(domNode, element);
+
+        this._table = element.append("table");
+        this._tbody = this._table.append("tbody");
+    }
+    update(domNode, element) {
+        super.update(domNode, element);
+        const trSelection = this._tbody.selectAll("tr").data(this.data());
```
Comment (GordonSmith) [inline packages/html/src/SimpleTable.ts]: Move to below table.style (IOW keep with trSelection.enter())

File: packages/html/src/SimpleTable.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            .each(function(d) {
+                const tr = d3Select(this);
+                const tdSelection = tr.selectAll("td").data(d);
+                tdSelection.enter()
+                    .append("td")
+                    .text(_d => (_d).toString())
+                    ;
+                tdSelection.exit().remove();
+            })
+            ;
```
Comment (GordonSmith) [inline packages/html/src/SimpleTable.ts]: Missing `update` and `exit` sections - won't work when data changes...

File: packages/html/src/SimpleTable.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            .each(function(d) {
+                const tr = d3Select(this);
+                const tdSelection = tr.selectAll("td").data(d);
+                tdSelection.enter()
+                    .append("td")
+                    .each(function(n, i) {
+                        d3Select(this)
+                            .classed("col-" + i, true)
+                            ;
+                    })
```
Comment (GordonSmith) [inline packages/html/src/SimpleTable.ts]: Does not need to be inside an "each"

File: packages/html/src/SimpleTable.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { HTMLWidget } from "@hpcc-js/common";
+import { select as d3Select } from "d3-selection";
+
+export class SimpleTable extends HTMLWidget {
+    protected _table;
+    protected _tbody;
+    _transformedData;
```
Comment (GordonSmith) [inline packages/html/src/SimpleTable.ts]: Looks dubious - remove.

Comment (jbrundage) [reply packages/html/src/SimpleTable.ts]: It's used to allow extending classes to transform `this.data()` before calling update...

Comment (GordonSmith) [reply packages/html/src/SimpleTable.ts]: Where (and why)?

Comment (jbrundage) [reply packages/html/src/SimpleTable.ts]: In the `BreakdownTable.ts` (it wasn't part of the PR until I amended it this morning)

Comment (GordonSmith) [reply packages/html/src/SimpleTable.ts]: This is not referenced anywhere in this PR, it should be removed (if needed in dependent code, we can review it then).

**Comment by GordonSmith** [comment]
@jbrundage - two comments above.

File: packages/html/src/SimpleTable.ts (hunk 4 above)
Comment (jbrundage) [reply packages/html/src/SimpleTable.ts]: I've replaced it with this method. Does that work? That will allow extending widgets to overwrite that method with their own transformation logic.
```javascript
    transformData(data) {
        return data;
    }
```

Comment (GordonSmith) [reply packages/html/src/SimpleTable.ts]: Add that method in your other PR, when its needed - otherwise your just adding a "Nothing" method.

Comment (GordonSmith) [reply packages/html/src/SimpleTable.ts]: As far as I can tell - even if an extending method overrode it, it has no use in this class?

File: packages/html/src/SimpleTable.ts (hunk 3 above)
Comment (GordonSmith) [reply packages/html/src/SimpleTable.ts]:
```
                tdSelection.enter()
                    .append("td")
                    .attr("class", (d, i) ->"col-" + i)
...
```
Should do the same thing.

File: packages/html/src/SimpleTable.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            ;
+        const theadTrSelection = this._thead.selectAll("tr > th").data(this.columns());
+        theadTrSelection.enter()
+            .append("th")
+            .attr("class", (n, i) => `th-${i}`)
+            .merge(theadTrSelection)
+            .text(_d => (_d).toString())
+            ;
+        theadTrSelection.exit().remove();
+        const trSelection = this._tbody.selectAll("tr").data(this.transformData(this.data()));
```
Comment (jbrundage) [inline packages/html/src/SimpleTable.ts]: @GordonSmith it's used here

Comment (jbrundage) [reply packages/html/src/SimpleTable.ts]: Nevermind, it's removed

File: packages/html/src/SimpleTable.ts (hunk 4 above)
Comment (jbrundage) [reply packages/html/src/SimpleTable.ts]: Done, it's removed

Comment (GordonSmith) [reply packages/html/src/SimpleTable.ts]: `_transformedData` is still there...

**Pull request merged**